### ISSUE TITLE
Make CurveTween curve a regular parameter

### ISSUE THREAD

**Issue by sir-boformer** [flutter/flutter#21433]
Right now the `CurveTween` constructor looks like this:

```dart
  /// Creates a curve tween.
  ///
  /// The [curve] argument must not be null.
  CurveTween({ @required this.curve })
    : assert(curve != null);
```

Usage currently looks like this:
```dart
CurveTween(curve: Curves.easeInOut);
```

There is a single required parameter, and I think it's quite certain that a `CurveTween` needs a `Curve` to function.

I propose to make it a regular parameter to make usage easier and a bit more fail-safe:

```dart
CurveTween(Curves.easeInOut);
```

If this is approved by the Flutter team, I will create a PR.

**Comment by zoechi** [comment]
@Hixie

**Comment by Hixie** [comment]
I agree that this would make sense, but unfortunately it would be a breaking change, so it's probably too late to really make this change.

**Comment by Hixie** [comment]
FWIW, the original thinking here was that we might add a "reverseCurve" argument, similar to how CurvedAnimation works. We haven't done that yet but it's still possible that we should one day do that.

**Comment by Hixie** [comment]
The problem isn't one small breaking change, it's 1,001 small breaking changes. If we make this change, we'd also have to make a thousand other breaking changes of the same nature, and eventually everyone would be spending all their time just handling our breaking changes.

**Comment by adil192** [comment]
Should this be closed now that we have dot shorthands? (https://github.com/flutter/flutter/pull/166012#pullrequestreview-2788341143)

Edit: No, the code sample in the linked comment isn't (currently) supported since `Curve` and `Curves` are completely distinct to Dart.
